### PR TITLE
[MM-59292] Add metadata to Support Packet

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary

Example:
```yaml
version: 1
type: support-packet
generated_at: 1720184583187
server_version: 9.11.0
server_id: ciu8wqr7mbbqjb74cao5t6a5gh
license_id: ntisr7wfwbghpyakh87fazbqma
customer_id: p9un369a67gimj4yd6i6ib39wh
```

#### Ticket Link
https://mattermost.atlassian.net/issues/MM-59292

#### Release Note
```release-note
Add metadata to Support Packet
```

### CODE REVIEW

**Comment by hanzei** [comment]
@isacikgoz Do you have any ideas why the generated metadata could have an empty `ServerVersion` field? `{"timestamp":"2024-07-05 13:20:50.480 Z","level":"error","msg":"Failed to generate file for support package","caller":"app/support_packet.go:55","error":"failed to generate packet metadata: invalid metadata: could not parse server version: strconv.ParseUint: parsing \"\": invalid syntax","file":"metadata"}` (https://github.com/mattermost/mattermost/runs/27086051667)

**Comment by hanzei** [comment]
Figured it out. Another test writes to `model.CurrentVersion`. Fixed that via https://github.com/mattermost/mattermost/pull/27591. CI will be broken until the other PR is merged.

@isacikgoz @pvev Please take a look.

**Comment by hanzei** [comment]
https://github.com/mattermost/mattermost/pull/27591 got merged. CI should be green now.

File: server/channels/app/support_packet_test.go [hunk 1]
```diff
@@ -394,3 +397,20 @@ func TestCreateSanitizedConfigFile(t *testing.T) {
 	assert.Positive(t, len(fileData.Body))
 	assert.NoError(t, err)
 }
+
+func TestCreateSupportPacketMetadata(t *testing.T) {
```
Comment (pvev) [inline server/channels/app/support_packet_test.go:401]: nit: maybe add test to cover error in metadata generation and error in YAML marshalling?

Comment (hanzei) [reply server/channels/app/support_packet_test.go:401]: I did find it hard to add a test for the error case as `createSupportPacketMetadata` only calls `model.GeneratePacketMetadata`. Surely, it's possible to introduce dependency injection and replace the implementation of `model.GeneratePacketMetadata` with something else, but I don't think that added complexity is worth the additional test coverage.

Do you have other suggestions on how to test the error case?

**Comment by hanzei** [comment]
@DHaussermann Heads up that I'm going to defer QA review for this PR as it would like to merge it for the next ESR 9.11.

Tests steps are in the ticket.

**Pull request merged**